pkg/network/network.go

```diff
@@ -20,12 +20,18 @@ import (
 	"encoding/binary"
 	"fmt"
 	"net"
+	"sync"
+	"time"
 
 	"github.com/pkg/errors"
 	"k8s.io/klog/v2"
 )
 
+const defaultReservationPeriod = 1 * time.Minute
+
 var (
+	reservedSubnets = sync.Map{}
```

Comment:
how about using our Existing lock package ?
```
func PathMutexSpec(path string) mutex.Spec {
````
https://github.com/medyagh/minikube/blob/f95d43a2c78070ab10a4e7d134ff2d0952a5ca86/pkg/util/lock/lock.g o#L48

for better maintainability of the lock code, what is the Special about Locking a network Name that is different than other Locks in minikube?

Reply:
here we need to lock a specific currently free network segment (not a network name) for a caller for a short period of time:

when a call to network.FreeSubnet() is made, it checks if a subnet is taken at the system level and skip those;
furthermore, in case of concurrent calls to network.FreeSubnet(), it also needs to skip all those network segments that were previously returned as 'free' (to other callers), but those other callers, within a configurable grace period, have not yet allocated the subnet at the system level

the existing locking mechanism with lock.PathMutexSpec() takes care of concurrent requests that would want to write to the same file by making them ***wait*** for their turn

on the other hand, with the free subnet allocation, we need a mechanism that would ***instantly know & skip*** if a subnet is either already taken or 'reserved' but not yet expired, and so to move on to the next free subnet

therefore i used sync.Map that has a built-in mechanism for safe concurrent access for 'reservations' with key:subnet and value:createdAt to be able to check if the subnet reservation expired (so free to reuse)

does that make sense?